comms/dht/src/dht.rs

```diff
@@ -174,7 +173,11 @@ impl Dht {
             .layer(store_forward::ForwardLayer::new(
                 Arc::clone(&self.peer_manager),
                 self.outbound_requester(),
-            ))
+            ));
+
+        // TODO: Adding these layers in Windows build create stack overflow in Tokio, to be fixed
```

Comment:
nit:
```suggestion
        // FIXME: The store and forward layers are excluded for Windows builds due to an unresolved stack overflow issue (#1416)
```

Having played around with this, I don't think the issue is with tokio per se - it's more likely something to do with the complexity of the `Layer` type returned form this function and its interaction with the stack on windows. I believe one solution will be box this type (i.e move it onto the heap). An audit of what is allocated on the stack vs heap for each layer might reveal the issue.

Reply:
Just a note from playing with this, it overflows regardless of which layers are added - seems to have more to do with the number of layers